File: advent_of_code/2023/day08/ts_with_deno/aoc20231208/main_test.ts
import { assertEquals } from "@std/assert";
import { solvePart1, solvePart2, Children, Input, parseInput } from "./main.ts";

Deno.test("Testing the parsing of a first example input data", function parseInputTest() {
  const expectedNodes = new Map<string, Children>();
  expectedNodes.set("AAA", {L:"BBB", R:"CCC"});
  expectedNodes.set("BBB", {L:"DDD", R:"EEE"});
  expectedNodes.set("CCC", {L:"ZZZ", R:"GGG"});
  expectedNodes.set("DDD", {L:"DDD", R:"DDD"});
  expectedNodes.set("EEE", {L:"EEE", R:"EEE"});
  expectedNodes.set("GGG", {L:"GGG", R:"GGG"});
  expectedNodes.set("ZZZ", {L:"ZZZ", R:"ZZZ"});
  const expected : Input = {
    instructions : ["R","L"],
    nodes: expectedNodes
  }
  const actual : Input = parseInput("./data/input_test1.txt");
  assertEquals(actual, expected)
});

Deno.test("Testing the parsing of a second example input data", function parseInputTest() {
  const expectedNodes = new Map<string, Children>();
  expectedNodes.set("AAA", {L:"BBB", R:"BBB"});
  expectedNodes.set("BBB", {L:"AAA", R:"ZZZ"});
  expectedNodes.set("ZZZ", {L:"ZZZ", R:"ZZZ"});
  const expected : Input = {
    instructions : ["L","L", "R"],
    nodes: expectedNodes
  }
  const actual : Input = parseInput("./data/input_test2.txt");
  assertEquals(actual, expected)
});

Deno.test("Resolving part1 with the first input data", function solvePart1Test() {
  const expected = 2;
  const actual = solvePart1("./data/input_test1.txt");
  assertEquals(actual, expected);
});

Deno.test("Resolving part1 with the second input data", function solvePart1Test() {
  const expected = 6;
  const actual = solvePart1("./data/input_test2.txt");
  assertEquals(actual, expected);
});

Deno.test("Resolving part2 with the dedicated input data", function solvePart2Test() {
  const expected = 6;
  const actual = solvePart2("./data/input_test_for_part2.txt");
  assertEquals(actual, expected);
});

File: advent_of_code/2025/day11_python/main_test.py
from main import extract_data, solve_part1, solve_part2, solve_part2


def test_extract_data():
    filename = "./data/input_test1.txt"
    # expected_input = [
    #     (".##.", [[3], [1, 3], [2], [2, 3], [0, 2], [0, 1]], (3, 5, 4, 7)),
    #     (
    #         "...#.",
    #         [[0, 2, 3, 4], [2, 3], [0, 4], [0, 1, 2], [1, 2, 3, 4]],
    #         (7, 5, 12, 7, 2),
    #     ),
    #     (
    #         ".###.#",
    #         [[0, 1, 2, 3, 4], [0, 3, 4], [0, 1, 2, 4, 5], [1, 2]],
    #         (10, 11, 11, 5, 10, 5),
    #     ),
    # ]
    graph = extract_data(filename)
    print("input:", graph)
    # assert input == expected_input


def test_solve_part1():
    filename = "./data/input_test1.txt"
    assert solve_part1(filename) == 5


def test_solve_part2():
    filename = "./data/input_test2.txt"
    assert solve_part2(filename) == 2


def test_solve_part2_bis():
    filename = "./data/input_test2.txt"
    assert solve_part2(filename) == 2
